### DISCUSSION TITLE
Easy way to get the input element inside the Upload component?

### DISCUSSION THREAD

**Discussion by VictorNorman** [reflex-dev/reflex#3697] [Q&A]
We need to allow the user to select directories, not files, on the client side. The only thing that has to be done is add the "webkitdirectory" attribute to the <input> tag that is used in the Upload component. I was hoping that one could use a useRef() hook easily to get a handle on the input element and then set the attribute on the element. Is there an easy way to do this?

**Comment by VictorNorman** [comment]
We found a solution: in the `rx.upload()` call, we put this:

```on_mount=rx.call_script("""document.querySelector("div#upload2 > input").setAttribute("webkitdirectory", "true");"""),```

the `id` is "upload2".

One weird thing though: that code is called *twice* .  (We tried using `toggleAttribute()`, but because the code is called twice, it toggle the attribute on and then off again).

Anyone know why the code is called twice?

**Reply by masenf** [reply]
`on_mount` is called twice in dev mode due to React StrictMode being enabled and running all `useEffect` twice
